### ISSUE TITLE
Comments: Edit Comment Screen Pulls HTTP links instead of HTTPS

### ISSUE THREAD

**Issue by tmmbecker** [Automattic/jetpack#7672]
This is pretty obscure. When clicking the **Edit** link on a comment, the Edit Comment screen is showing the HTTP link to the commenters primary URL rather than the HTTPS version.

#### Steps to reproduce the issue
1. Have other users with a primary HTTPS site connected to their username comment on any post.
2. Click the **Edit** link to edit the comment.
3. Note that the URL in the Edit Comment page is listed as HTTP. 

I had 3 testers comment - one site URL came through as HTTPS but the other two were not. 

Internal ref: 591844-hc

**Comment by kraftbj** [comment]
Specifically, was this with Jetpack Comments on? What were the various URLs attempted? I wasn't able to duplicate this on a first run but not sure what to check.

**Comment by tmmbecker** [comment]
Yes. This was with Jetpack Comments on. I had Richard, Carolyn and Paul leave comments. The first 2 URLs came through as HTTP (http://richardmtl.ca and http://learn.wordpress.com) but the last was correct.

I'm happy to give you admin creds to my site so that you can see if you'd like.

**Comment by kraftbj** [comment]
@tmmbecker That would be helpful -- DM to me whenever you have a chance. Thank you!
